### ISSUE TITLE
Incorrect "Expression is always false" with `long.operator %`

### ISSUE THREAD

**Issue by fuglede** [dotnet/roslyn#59680]
**Version Used**: 
Checked this in both Visual Studio 2019 and Visual Studio 2022

**Steps to Reproduce**:
Write the following code:

```cs
public static TimeSpan RoundTimeSpanUp(TimeSpan span, TimeSpan roundingTimeSpan)
{
    long remainder = span.Ticks % roundingTimeSpan.Ticks;
    return remainder == 0 ? span : new TimeSpan(span.Ticks - remainder + roundingTimeSpan.Ticks);
}
```
 

**Expected Behavior**:
No warnings are produced.

**Actual Behavior**:
The `remainder == 0` is underlined with a warning of "Expression is always false". The expression is, however, `true` in some cases, for instance when the input `TimeSpan` are identical.

**Comment by Youssef1313** [comment]
@fuglede What's the diagnostic ID you got? CA1508? If so, it belongs to dotnet/roslyn-analyzers.

**Comment by fuglede** [comment]
Thanks @Youssef1313. I created this issue to hastily; the issue was with ReSharper ([issue on their tracker](https://youtrack.jetbrains.com/issue/RSRP-488083)).
